### ISSUE TITLE
cx_freeze doesn't handle dynamic import on RNS package

### ISSUE THREAD

**Issue by ThomasFreedman** [marcelotduarte/cx_Freeze#2041]
**Prerequisite**
checked

**Describe the bug**
There is a set of imports determined at run time by the RNS package that cx_freeze doesn't handle well. I discussed this with [marcelotduarte](https://github.com/marcelotduarte) who suggested I report this issue (see [forum post](https://github.com/marcelotduarte/cx_Freeze/discussions/2039#discussioncomment-6970003)). Here is the code from RNS.Reticulum.py I changed that avoids the problem:

```
from .vendor.platformutils import get_platform

if get_platform() == "android":
    from .Interfaces import Interface
    from .Interfaces import LocalInterface
    from .Interfaces import AutoInterface
    from .Interfaces import TCPInterface
    from .Interfaces import UDPInterface
    from .Interfaces import I2PInterface
    from .Interfaces.Android import RNodeInterface
    from .Interfaces.Android import SerialInterface
    from .Interfaces.Android import KISSInterface
else:
    from .Interfaces import *
```
If I replace all of that with `from .Interfaces import *` everything works properly.

**To Reproduce**
See the forum thread I mentioned above for example code and further details.

**Expected behavior**
No run time import errors like this:
```
    RNS.Reticulum(self.RetCfgFile)    # Initialize the Reticulum Network Stack
  File "/volumes/Accounts/owner/PycharmProjects/pBoxV2/MediaCaptureForm/test/lib/python3.9/site-packages/RNS/Reticulum.py", line 255, in __init__
    self.__apply_config()
  File "/volumes/Accounts/owner/PycharmProjects/pBoxV2/MediaCaptureForm/test/lib/python3.9/site-packages/RNS/Reticulum.py", line 391, in __apply_config
    interface_mode = Interface.Interface.MODE_FULL
NameError: name 'Interface' is not defined
```

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - MX-Linux 21 (debian derrivaative)
 - OS architecture: amd64
 - cx_Freeze version: 6.15
 - Python version: 3.9.2

**Additional context**

**Comment by marcelotduarte** [comment]
You can test the patch in the latest development build:
`pip install --upgrade --pre --extra-index-url https://marcelotduarte.github.io/packages/ cx_Freeze`
For conda-forge the command is:
`conda install -y --no-channel-priority -S -c https://marcelotduarte.github.io/packages/conda cx_Freeze`

In this development version, the support for appimage is available too, see #2054

**Comment by ThomasFreedman** [comment]
Sorry I couldn't get back to this to provide feedback before you closed the issue. No worries. It may be another day or 2 but I WILL get back to it. I appreciate your creating the RNS hook and the appImage beta test code, both are greatly appreciated!

**Comment by marcelotduarte** [comment]
I will release a version soon and then include this hook, as there is no danger of impacting other things. If it is incomplete, we will reopen this issue and make a new PR, according to your feedback.

> I got one of my 2 GUI apps [...]  the 2nd app still has a problem with RNS despite my change to eliminate the dynamic import issue that caused the problem in the first app. [...]

I patched not only Interface, as I noticed a strange pattern when adding each module that would not work with cx_Freeze.

**Comment by marcelotduarte** [comment]
[Release 6.15.8](https://github.com/marcelotduarte/cx_Freeze/releases/tag/6.15.8) is out!

**Comment by ThomasFreedman** [comment]
Thank you so much @marcelotduarte for such a fast response for this issue AND the AppImage addition you're working on for Linux. I am very happy to report not only does the hook you created for the RNS module to resolve the dynamic (run time) import problem, I have also tested the AppImage creation you generously gave me access to test and it works perfectly! I'm sure there are others anxiously awaiting the inclusion of AppImage support for Linux, and don't forget to mention the RNS hook in your next release.

I also got the app working under pyinstaller, however it suffers from the same dynamic import problem I reported here. The size of the AppImage is almost the exact same size as the binary produced by pyinstaller.

Both cx_freeze and pyinstaller seem to share much in common - both have build and dist folders and both use "hooks", tho not sure if the hooks are compatible between the two tools. Both also use a similar method to reference files relocated at run time. It was trivial to modify the find_data_file method to switch between the two programs. 

For me I'm going with the AppImage produced by cx_freeze given the most excellent support and clean results produced.

Great work sir!
